typed/merge_test.go

```diff
@@ -345,19 +345,19 @@ var mergeCases = []mergeTestCase{{
 	}, {
 		`{"list":[{"key":"a","id":1,"value":{"a":"a"}}]}`,
 		`{"list":[{"key":"a","id":2,"value":{"a":"a"}}]}`,
-		`{"list":[{"key":"a","id":2,"value":{"a":"a"}},{"key":"a","id":1,"value":{"a":"a"}}]}`,
+		`{"list":[{"key":"a","id":1,"value":{"a":"a"}},{"key":"a","id":2,"value":{"a":"a"}}]}`,
 	}, {
 		`{"list":[{"key":"a","id":1},{"key":"b","id":1}]}`,
 		`{"list":[{"key":"a","id":1},{"key":"a","id":2}]}`,
-		`{"list":[{"key":"a","id":1},{"key":"a","id":2},{"key":"b","id":1}]}`,
+		`{"list":[{"key":"a","id":1},{"key":"b","id":1},{"key":"a","id":2}]}`,
 	}, {
 		`{"list":[{"key":"b","id":2}]}`,
 		`{"list":[{"key":"a","id":1},{"key":"b","id":2},{"key":"c","id":3}]}`,
 		`{"list":[{"key":"a","id":1},{"key":"b","id":2},{"key":"c","id":3}]}`,
 	}, {
 		`{"list":[{"key":"a","id":1},{"key":"b","id":2},{"key":"c","id":3}]}`,
 		`{"list":[{"key":"c","id":3},{"key":"b","id":2}]}`,
-		`{"list":[{"key":"c","id":3},{"key":"b","id":2},{"key":"a","id":1}]}`,
+		`{"list":[{"key":"a","id":1},{"key":"c","id":3},{"key":"b","id":2}]}`,
```

Comment:
Can we include some of the complicated examples we've come up with recently? i.e. https://gist.github.com/apelisse/a7073ba994e1de15c1d11c79cfabdb0e#file-merge-py-L66-L78

Reply:
Feel free to test with `setStr` above.

Reply:
Actually it'd be nice to see if we merge properly even in some of these complex use-cases.

Reply:
+1 to including the complex use cases, though I'm not sure this is what I'd expect:

```
        self.assertEqual(jordan_merge("abcdefghij", "1b2h3e4kl"), "a1bcd2h3efgij4kl")
```

Reply:
if we always honor RHS order, maximize honoring LHS order when possible, and append new items, I thought we'd end up with:

```
        self.assertEqual(jordan_merge("abcdefghij", "1b2h3e4kl"), "a1bcdfg2hij3e4kl")
```

Reply:
`a1bcd2h3efgij4kl` and `a1bcdfg2hij3e4kl` violate the same number of partial orders defined in LHS.
former violations: h > e, h > f, h >g
latter violations: e < h, e < i, e < j

Reply:
The "appending" is still a problem, though. Let me think about it too.